pkg/artifactory/resource/configuration/resource_artifactory_backup.go

```diff
@@ -120,17 +120,18 @@ func ResourceArtifactoryBackup() *schema.Resource {
 	}
 
 	var resourceBackupRead = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-		backups := &Backups{}
-		backup := unpackBackup(d)
+		data := &util.ResourceData{ResourceData: d}
+		key := data.GetString("key", false)
 
+		backups := &Backups{}
```

Comment:
On my PR for the user datasource we saw that we should not make this a reference, but the object itself. Is the reasoning different here, or should this also be an object?

Reply:
No different. This is also wrong. This is why we strive to not commit bad code because others will copy/paste them.

Reply:
I will fix them 😄